### PR TITLE
Improve failure behaviour from ert client to storage

### LINKED ISSUE #2953: `ert3 init` leaves directory `.ert` behind when initialization fails
**Describe the bug**
If an ert3 workspace is attempted initialized, but connection with the storage server fails, the directory `.ert` is still created and rerunning `ert3 init` (after storage is started) will not work until the directory is manually removed.

**To Reproduce**
1. terminal A: `ert3 init`. Results possibly in `RuntimeError: Server started, but none of the URLs provided worked` when it has tried using URL from an outdated `storage_server.json`.
2. terminal B: `ert3 service start storage` (and ensure the wrong `storage_server.json` is overwritten or overridden)
3. terminal A: `ert3 init`. Results in `Already inside an ERT workspace, found /work/projects/ert/ert3_examples/polynomial/.ert`

**Expected behavior**
Successful initialization in step 3.

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #2953 

**Approach**
The console already has functionality for removing the `.ert` directory, but only if it catches a TimeoutError. The storage initialization raises a RuntimeError. I chose to change the exception raised instead of catching one  more exception.

https://github.com/equinor/ert/blob/main/ert3/console/_console.py#L223-L229

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

**Comment by berland** [comment]
Bonus commit added that handles a similar situation when the workspace already exists in storage.

**Review by sondreso — APPROVED** [review]
LGTM!

File: ert/storage/_storage.py [hunk 1]
```diff
@@ -535,7 +535,9 @@ def init(*, workspace_name: str) -> None:
 
     for special_key in _SPECIAL_KEYS:
         if f"{workspace_name}.{special_key}" in experiment_names:
-            raise ValueError("Storage already initialized")
+            raise ValueError(
```
Comment (xjules) [inline ert/storage/_storage.py:538]: Just a small detail here. Shouldn't this be a different error: eg. `NameError`?

Comment (jondequinor) [reply ert/storage/_storage.py:538]: `NameError` is very specific, maybe `RuntimeError` is a better match?

Comment (xjules) [reply ert/storage/_storage.py:538]: `RuntimeError` is another option and is indeed more general.

**Comment by berland** [comment]
Tests added. Is it ok to expose the ".ert" implementation detail in tests?

**Review by sondreso — APPROVED** [review]
I think the usage of ".ert" in the test is ok in this scenario.

**Pull request merged**